packages/@tailwindcss-upgrade/src/migrate-postcss.ts

```diff
@@ -0,0 +1,287 @@
+import fs from 'node:fs/promises'
+import path from 'node:path'
+import { pkg } from './utils/packages'
+import { info, success, warn } from './utils/renderer'
+
+// Migrates simple PostCSS setups. This is to cover non-dynamic config files
+// similar to the ones we have all over our docs:
+//
+// ```js
+// module.exports = {
+//   plugins: {
+//     'postcss-import': {},
+//     'tailwindcss/nesting': 'postcss-nesting',
+//     tailwindcss: {},
+//     autoprefixer: {},
+//   }
+// }
+export async function migratePostCSSConfig(base: string) {
+  let didMigrate = false
+  let didAddPostcssClient = false
+  let didRemoveAutoprefixer = false
+  let didRemovePostCSSImport = false
+
+  // Priority 1: Handle JS config files
+  let jsConfigPath = await detectJSConfigPath(base)
+  if (jsConfigPath) {
+    let result = await migratePostCSSJSConfig(base, jsConfigPath)
+
+    if (result) {
+      didMigrate = true
+      didAddPostcssClient = result.didAddPostcssClient
+      didRemoveAutoprefixer = result.didRemoveAutoprefixer
+      didRemovePostCSSImport = result.didRemovePostCSSImport
+    }
+  }
+
+  // Priority 2: Handle package.json config
+  let packageJson
+  try {
+    packageJson = JSON.parse(await fs.readFile(path.resolve(base, 'package.json'), 'utf-8'))
+  } catch {}
+  if (!didMigrate && packageJson && 'postcss' in packageJson) {
+    let result = await migratePostCSSPackageJsonConfig(base, packageJson)
+
+    if (result) {
+      didMigrate = true
+      didAddPostcssClient = result.didAddPostcssClient
+      didRemoveAutoprefixer = result.didRemoveAutoprefixer
+      didRemovePostCSSImport = result.didRemovePostCSSImport
+    }
+  }
+
+  if (!didMigrate) {
+    info(`No PostCSS config found, skipping migration.`)
+    return
+  }
+
+  if (didAddPostcssClient) {
+    try {
+      await pkg('add -D @tailwindcss/postcss@next', base)
+    } catch {}
+  }
+  if (didRemoveAutoprefixer) {
+    try {
+      await pkg('remove autoprefixer', base)
+    } catch {}
+  }
+  if (didRemovePostCSSImport) {
+    try {
+      await pkg('remove postcss-import', base)
+    } catch {}
+  }
```

Comment:
We should merge these two `remove` calls into one command given that npm add/remove can sometimes take a decent chunk of time.

As far as I know every package manager supports removing more than one package